lib/agent0/interactive_fuzz/fuzz_long_short_maturity_values.py

```diff
@@ -0,0 +1,191 @@
+min_trade = interactive_hyperdrive.hyperdrive_interface.pool_config.minimum_transaction_amount
+trade_list: list[tuple[InteractiveHyperdriveAgent, HyperdriveActionType, FixedPoint]] = []
+for agent_index in range(NUM_TRADES):  # 1 agent per trade
+    budget = FixedPoint(
+        scaled_value=int(np.floor(rng.uniform(low=min_trade.scaled_value * 10, high=int(1e23))))
+    )  # Give a little extra money to account for fees
+    agent = interactive_hyperdrive.init_agent(base=budget, eth=FixedPoint(100))
+    trade_type = rng.choice(available_actions, size=1)[0]
+    trade_amount_base = FixedPoint(
+        scaled_value=int(
+            rng.uniform(
+                low=min_trade.scaled_value,
+                high=int(
+                    budget.scaled_value / 2
+                ),  # Don't trade all of their money, to make sure they have enough for fees
+            )
+        )
+    )
+    trade_list.append((agent, trade_type, trade_amount_base))
+
+# %%
+# List of columns in pool info to check between the initial pool info and the latest pool info.
+check_columns = [
+    "share_reserves",
+    "shorts_outstanding",
+    "withdrawal_shares_proceeds",
+    "share_price",
+    "long_exposure",
+    # Added in addition to original script
+    "bond_reserves",
+    "lp_total_supply",
+    "longs_outstanding",
+]
+
+
+# %%
+# Open some trades
+# TODO: include add liquidity
+trade_events: list[tuple[InteractiveHyperdriveAgent, OpenLong | OpenShort]] = []
+for trade in trade_list:
+    agent, trade_type, trade_amount = trade
+    if trade_type == HyperdriveActionType.OPEN_LONG:
+        trade_event = agent.open_long(base=trade_amount)
+    elif trade_type == HyperdriveActionType.OPEN_SHORT:
+        trade_event = agent.open_short(bonds=trade_amount)
+    else:
+        raise AssertionError(f"{trade_type=} is not supported.")
+    trade_events.append((agent, trade_event))
+
+# automatically created by sending transactions
+starting_checkpoint = interactive_hyperdrive.hyperdrive_interface.current_pool_state.checkpoint
+
+# %%
+# snapshot the chain, so we can load the snapshot & close in different orders
+chain.save_snapshot()
+
+check_data: dict | None = None
+# %%
+# advance the time to at least the position duration, maximum of two position durations.
+
+position_duration = interactive_hyperdrive.hyperdrive_interface.pool_config.position_duration
+rng = np.random.default_rng()  # No seed, we want this to be random every time it is executed
+extra_time = int(np.floor(rng.uniform(low=0, high=position_duration)))
+
+# advances time and mines a block
+current_time = interactive_hyperdrive.hyperdrive_interface.current_pool_state.block_time
+checkpoint_duration = interactive_hyperdrive.hyperdrive_interface.pool_config.checkpoint_duration
+
+starting_checkpoint_time = current_time - current_time % checkpoint_duration
+
+chain.advance_time(position_duration + 30)
+
+# create a checkpoint
```

Comment:
This can be done through `chain.advance_time(..., create_checkpoints = True)`, so you don't need the logic below to create checkpoints

Reply:
Although, the above method creates all intermediate checkpoints, which may be a bit of overkill. If this case happens more often than not, I may add that into interactive hyperdrive. If you want below to be a bit more clean, hyperdrive interface has a function to create a checkpoint at the last checkpoint time:

```
interactive_hyperdrive.hyperdrive_interface.create_checkpoint(sender_account)
```